src/main/kotlin/ar/edu/unahur/obj2/socios/Cliente.kt

```diff
@@ -1,4 +1,61 @@
 package ar.edu.unahur.obj2.socios
 
-class Cliente {
+class Cliente(var estadoQueManeja: EstadoDeAnimo, var plataEnBolsillo: Double, var residencia: LugarDondeVive) {
+    fun propinaSegunEstadoAnimo(valorPedido: Double) = estadoQueManeja.cuantoDejaDePropina(plataEnBolsillo, valorPedido)
+
+    fun propinaSegunZona(valorPedido: Double) = residencia.modificacionDePropinaPorLugar(this.propinaSegunEstadoAnimo(valorPedido))
+}
+
+abstract class EstadoDeAnimo{
+    abstract fun cuantoDejaDePropina(plataEnBolsillo: Double, valorPedido: Double) : Double
+}
+
+object estaEnojado : EstadoDeAnimo() {
+    override fun cuantoDejaDePropina(plataEnBolsillo: Double, valorPedido: Double) : Double {
+        return 0.0
+    }
+}
+
+object estaIndiferente : EstadoDeAnimo() {
+    override fun cuantoDejaDePropina(plataEnBolsillo: Double, valorPedido: Double) : Double {
+        return plataEnBolsillo
+    }
+}
+object estaFeliz : EstadoDeAnimo() {
+    override fun cuantoDejaDePropina(plataEnBolsillo: Double, valorPedido: Double): Double {
+        return valorPedido * 1.25
```

Comment:
> le deja como propina el 25% de lo que salió el pedido

El código que hicieron le deja el 125%, no el 25%. Sería así:
```suggestion
        return valorPedido * 0.25
```